db.go

```diff
@@ -501,7 +506,7 @@ func (db *DB) close() (err error) {
 	// and remove them completely, while the block / memtable writer is still
 	// trying to push stuff into the memtable. This will also resolve the value
 	// offset problem: as we push into memtable, we update value offsets there.
-	if !db.mt.sl.Empty() {
+	if !db.opt.ReadOnly {
```

Comment:
We might end up pushing an empty memtable. Why push it?

Reply:
`handleFlushTask` do nothing on empty memtable. Also, DecrRef() is called hence deleting the empty memtable. So, I am avoiding special handling here.

Reply:
done